commcare/src/main/resources/velocity.templates/task-data-provider.vm

```diff
@@ -1,13 +1,14 @@
-#macro( buildDisplayName $subject $name $configName )
-    "$esc.java($DisplayNameHelper.buildDisplayName($subject, $name, $configName))"
+#macro( buildDisplayName $subject $name $applicationName $configName )
+    "$esc.java($DisplayNameHelper.buildDisplayName($subject, $name, $applicationName, $configName))"
 #end
 
 {
     "name": "commcare",
     "objects": [
             #foreach($config in $configurations)
+              #foreach ($applicationName in $configToApplications.get($config.getConfigName()))
```

Comment:
This loop here is incorrect and generates duplicate lookup objects. We do not want to create separate lookup object for each application (form/case belongs to one, certain application - not to all of them). The number of objects should remain the same. The only thing that should be changed is that for each form/case etc. the application name this instance belongs to should be passed to the DisplayNameHelper.